hnn_qt5.py

```diff
@@ -1016,7 +1136,421 @@ def addDist (self):
+        tab.layout.addWidget(QLineEdit("50.0"), row, 4)
+        tab.layout.addWidget(QLabel(), row, 5)
+      else:
+        tab.layout.addWidget(QLabel("range (%)"), row, 3)
+        tab.layout.addWidget(QLineEdit("500.0"), row, 4)
+        tab.layout.addWidget(QLabel(), row, 5)
+      row += 1
+
+    # while row < 11:
+    #   tab.layout.addItem(QSpacerItem(tab.layout.sizeHint().width(),
+    #                                  self.dqline[k].sizeHint().height(),
+    #                                  vPolicy=QSizePolicy.Maximum),
+    #                                  row, 1, columnSpan=-1)
+    #   row += 1
+
+    tab.setLayout(tab.layout)
+
+  def addProx (self):
+    self.nprox += 1 # starts at 1
+    # evprox feed strength
+    dprox = OrderedDict([('t_evprox_' + str(self.nprox), 0.), # times and stdevs for evoked responses
+                         ('sigma_t_evprox_' + str(self.nprox), 2.5),
+                         #('numspikes_evprox_' + str(self.nprox), 1),
+                         ('gbar_evprox_' + str(self.nprox) + '_L2Pyr_ampa', 0.),
+                         ('gbar_evprox_' + str(self.nprox) + '_L2Pyr_nmda', 0.),
+                         ('gbar_evprox_' + str(self.nprox) + '_L2Basket_ampa', 0.),
+                         ('gbar_evprox_' + str(self.nprox) + '_L2Basket_nmda', 0.),
+                         ('gbar_evprox_' + str(self.nprox) + '_L5Pyr_ampa', 0.),
+                         ('gbar_evprox_' + str(self.nprox) + '_L5Pyr_nmda', 0.),
+                         ('gbar_evprox_' + str(self.nprox) + '_L5Basket_ampa', 0.),
+                         ('gbar_evprox_' + str(self.nprox) + '_L5Basket_nmda', 0.)])
+    self.ld.append(dprox)
+    self.addtransvarfromdict(dprox)
+    tab = self.addTab(True, 'evprox_' + str(self.nprox))
+    self.addGridToTab(dprox, tab)
+    self.addtips()
+
+  def addDist (self):
+    self.ndist += 1
+    # evdist feed strengths
+    ddist = OrderedDict([('t_evdist_' + str(self.ndist), 0.),
+                         ('sigma_t_evdist_' + str(self.ndist), 6.),
+                         #('numspikes_evdist_' + str(self.ndist), 1),
+                         ('gbar_evdist_' + str(self.ndist) + '_L2Pyr_ampa', 0.),
+                         ('gbar_evdist_' + str(self.ndist) + '_L2Pyr_nmda', 0.),
+                         ('gbar_evdist_' + str(self.ndist) + '_L2Basket_ampa', 0.),
+                         ('gbar_evdist_' + str(self.ndist) + '_L2Basket_nmda', 0.),
+                         ('gbar_evdist_' + str(self.ndist) + '_L5Pyr_ampa', 0.),
+                         ('gbar_evdist_' + str(self.ndist) + '_L5Pyr_nmda', 0.)])
+    self.ld.append(ddist)
+    self.addtransvarfromdict(ddist)
+    tab = self.addTab(True, 'evdist_' + str(self.ndist))
+    self.addGridToTab(ddist, tab)
+    self.addtips()
+
+  def runOptimization(self):
+    # update the opt info dict to capture num_sims from GUI
+    self.updateOptInfo()
+
+    # run the actual optimization
+    self.optrun_func()
```

Comment:
This comes from HNNGUI.startoptmodel():
https://github.com/jonescompneurolab/hnn/pull/59/files#diff-c1edfba3982d28d27b5d948841bb155fR2234

Passed to BaseParamDialog then finally OptEvokedInputParamDialog